Jet_fitting_3/include/CGAL/Monge_via_jet_fitting.h

```diff
@@ -758,14 +803,9 @@ void Monge_via_jet_fitting<DataKernel, LocalKernel, SvdTraits>::
 switch_to_direct_orientation(Vector_3& v1, const Vector_3& v2,
 			    const Vector_3& v3) 
 {
-  typedef typename CGAL::Linear_algebraCd<FT>::Matrix Matrix;
-  Matrix M(3,3);
-  for (int i=0; i<3; i++) M(i,0) = v1[i];
-  for (int i=0; i<3; i++) M(i,1) = v2[i];
-  for (int i=0; i<3; i++) M(i,2) = v3[i];
-
-  CGAL::Sign orientation = CGAL::Linear_algebraCd<FT>::sign_of_determinant(M);
-  if (orientation == CGAL::NEGATIVE) v1 = -v1;
+  Vector_3 v = cross_product(v2,v3);
+  double d = v*v1;
+  if(d < 0)v1 = -v1;
 }
 
```

Comment:
Don't we have a predicate are_in_the_same_halfspace (not exactly this name)?

andreas

On 06/08/2015 11:23, Sebastien Loriot wrote:
> In Jet_fitting_3/include/CGAL/Monge_via_jet_fitting.h
> <https://github.com/CGAL/cgal/pull/255#discussion_r36395596>:
>
>> @@ -758,14 +803,9 @@ void Monge_via_jet_fitting<DataKernel, LocalKernel, SvdTraits>::
>>  switch_to_direct_orientation(Vector_3& v1, const Vector_3& v2,
>>  			    const Vector_3& v3)
>>  {
>> -  typedef typename CGAL::Linear_algebraCd<FT>::Matrix Matrix;
>> -  Matrix M(3,3);
>> -  for (int i=0; i<3; i++) M(i,0) = v1[i];
>> -  for (int i=0; i<3; i++) M(i,1) = v2[i];
>> -  for (int i=0; i<3; i++) M(i,2) = v3[i];
>> -
>> -  CGAL::Sign orientation = CGAL::Linear_algebraCd<FT>::sign_of_determinant(M);
>> -  if (orientation == CGAL::NEGATIVE) v1 = -v1;
>> +  Vector_3 v = cross_product(v2,v3);
>> +  double d = v*v1;
>> +  if(d < 0)v1 = -v1;
>>  }
>>
>
> Here we have a function that was using a predicate that is now using a
> construction. If the underlying kernel does not have exact construction
> this might lead to errors.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/CGAL/cgal/pull/255/files#r36395596>.
>

-- 
Andreas Fabri, PhD
Chief Officer, GeometryFactory
Editor, The CGAL Project

phone: +33.492.954.912    skype: andreas.fabri